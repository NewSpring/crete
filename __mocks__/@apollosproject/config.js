const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    SHARE_URL: 'https://newspring.cc',
    TIMEZONE: 'America/New_York',
  },
  ROCK_CONSTANTS: {
    IMAGE: 10,
    AUDIO_FILE: 77,
    VIDEO_FILE: 79,
  },
  ROCK_MAPPINGS: {
    FEED_CONTENT_CHANNEL_IDS: [1, 2, 3, 4, 6, 8],
    SERIES_CONTENT_CHANNEL_TYPE_IDS: [6, 7],
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [18, 20],
      },
      DevotionalContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [17],
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
      WeekendContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
});

export default ApolloServer;
