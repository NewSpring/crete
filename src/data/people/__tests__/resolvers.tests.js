import { graphql } from 'graphql';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import { Group, Auth } from '@apollosproject/data-connector-rock';
import * as Person from '../index';
import oneRockPerson from '../../mocks/person';

const { getSchema, getContext } = createTestHelpers({ Person, Group, Auth });

describe('Person resolver', () => {
  const context = getContext();
  context.dataSources.Person.isStaff = () => true;
  context.dataSources.Group.getByPerson = () => [{ id: 1, name: 'Group1' }];
  context.dataSources.Auth.getCurrentPerson = () =>
    Promise.resolve(oneRockPerson);
  const schema = getSchema();

  it('gets a person', async () => {
    const query = `
      {
        currentUser {
          profile {
            id
            isGroupLeader
            isStaff
            photo {
              uri
            }
          }
        }
      }
    `;
    const result = await graphql(schema, query, {}, context);
    expect(result).toMatchSnapshot();
  });
});
