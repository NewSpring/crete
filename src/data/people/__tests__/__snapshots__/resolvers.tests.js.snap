// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campus gets campuses 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "Person:559b23fd0aa90e81b1c023e72e230fa1",
    },
  },
}
`;

exports[`Person resolver gets a person 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Cannot query field "isGroupLeader" on type "Node". Did you mean to use an inline fragment on "Person"?],
    [GraphQLError: Cannot query field "isStaff" on type "Node". Did you mean to use an inline fragment on "Person"?],
  ],
}
`;
